public/js/clients/firefox.js

```diff
@@ -100,16 +100,22 @@ function initPage(actions) {
     threadClient.addListener(eventName, clientEvents[eventName]);
   });
 
-  threadClient.reconfigure({
-    "useSourceMaps": false,
-    "autoBlackBox": false
+  // In Firefox, we need to initially request all of the sources. In
+  // the usual case, the debugger hasn't seen the sources before and
+  // will trigger `newSource` notifications for them. However, in a
+  // few cases these sources may already exist in the debugger server:
+  // other tools have interacted with them, a `debugger` statement was
+  // hit before the debugger is initialized, or a page was loading
+  // from bfcache. In all these cases, we need to make sure to fire
+  // `newSource` notifications even if the server doesn't. Otherwise
+  // they won't appear in the debugger.
+  return threadClient.getSources().then(({ sources }) => {
+    sources.forEach(source => clientEvents.newSource(null, { source }));
+
+    // If the threadClient is already paused, make sure to show a
+    // paused state.
+    clientEvents.paused(null, threadClient.getLastPausePacket());
```

Comment:
You still need to check if the `getLastPausePacket()` is null; the thread may not be paused, so there is none:

```js
    const pausedPacket = threadClient.getLastPausePacket();
    if(pausedPacket) {
      clientEvents.paused(null, pausedPacket);
    }
```